pynamodb/connection/base.py

```diff
@@ -338,12 +344,24 @@ def _make_api_call(self, operation_name, operation_kwargs):
         1. It's faster to avoid using botocore's response parsing
         2. It provides a place to monkey patch requests for unit testing
         """
+
+        """
+        Garrett TODO:
+         * why was botocore's response parsing slow? is that still the case?
```

Comment:
botocore uses an internal construct called "shapes" to validate responses meet the expected format. shapes are made up of other shapes. shapes are defined in JSON model files, which can generically be shared and updated among across their various SDKs. because of this, botocore does a bunch of recursive expansion and parsing to ultimately validate the response. we don't need any of the validation though.

botocore is designed for compatibility and maintainability rather than efficiency or speed.